sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/ConditionEvaluatesToConstant.cs

```diff
@@ -828,7 +828,13 @@ int ValueEquals(int i, int j)
 
         void DefaultExpression(object o)
         {
-            if (default(o) == null) { } // Compliant
+            if (default(o) == null) { } // Noncompliant {{Change this condition so that it does not always evaluate to 'true'.}}
+            int? amo = null;
+            if (amo == null) { } // Noncompliant
+            if (default(int?) == null) { } // Noncompliant
+
+            if (default(System.IO.FileAccess) != null) { } // Noncompliant
+            if (default(float) != null) { } // Noncompliant

```

Comment:
There's a missing case here: unconstrained generic type parameters are not checked.